Mathlib/SetTheory/Cardinal/Cofinality.lean

```diff
@@ -122,7 +128,7 @@ unbounded, in the sense `∀ a, ∃ b ∈ S, a ≤ b`.
 
 In particular, `cof 0 = 0` and `cof (succ o) = 1`. -/
 def cof (o : Ordinal.{u}) : Cardinal.{u} :=
-  o.liftOn (fun a ↦ Order.cof (swap a.rᶜ)) fun _ _ ⟨f⟩ ↦ f.compl.swap.cof_eq
+  o.liftOn _ fun _ _ ⟨f⟩ ↦ f.compl.swap.cof_eq
```

Comment:
I prefer to see the data-building argument in the source code even if Lean can find it from the proof by unification.

Reply:
Also please mention in the commit message that you're dropping some TC assumptions. Otherwise LGTM. Thanks!
bors d+